src/components/Camera/index.js

```diff
@@ -15,22 +15,55 @@ import PermissionsRecover from './Permissions/Recover'
 
 import classNames from 'classnames'
 import style from './style.css'
-import type { CameraPureType, CameraType, CameraActionType, CameraStateType} from './CameraTypes'
+import type { CameraPureType, CameraType, CameraActionType, CameraStateType, CameraErrorType} from './CameraTypes'
 import { checkIfWebcamPermissionGranted, parseTags } from '../utils'
 
-export const CaptureActions = ({handleClick, btnText, isFullScreen, btnClass}: CameraActionType) => {
+export const CaptureActions = ({handleClick, btnText, isFullScreen, btnClass, btnDisabled}: CameraActionType) => {
   return (
     <div className={style.captureActions}>
       <button
         className={classNames(style.btn, btnClass)}
-        onClick={handleClick}>
+        onClick={handleClick} disabled={btnDisabled}>
         <div className={classNames({[style.btnText]: isFullScreen})}>{btnText}</div>
       </button>
     </div>
   )
 }
 
-const reload = () => window.location.reload()
+class CameraError extends React.Component<CameraErrorType> {
+  fileInput = null
+
+  componentDidMount () {
+    this.props.trackScreen('camera_error')
+  }
+
+  handleUpload = () => {
+    if (this.fileInput) { this.props.onUploadFallback(this.fileInput.files[0]) }
+  }
+
+  onFallbackClick = () => { if (this.fileInput) { this.fileInput.click(); } }
+
+  errorInstructions = (text, onFallbackClick) =>
+    <span onClick={onFallbackClick} className={style.errorLink}>
+      { text }
+      <input type="file" id="fallback"
+        ref={(ref) => this.fileInput = ref} style={'display: none'}
```

Comment:
a bit inconsistent that some callbacks of of this class are passed as parameters (`this.onFallbackClick`) while others (`this.handleUpload`) are internally read. I would move this section to its own pure component